src/logging/request.js

```diff
@@ -0,0 +1,38 @@
+
+/**
+ * Module dependencies.
+ */
+
+import debugnyan from 'debugnyan';
+import { obfuscate } from './request-obfuscator';
+import request from 'request';
+import requestLogger from '@uphold/request-logger';
+
+/**
+ * Instances.
+ */
+
+const log = debugnyan('bitcoin-core:request');
+
+/**
+ * Default logging function.
+ */
+
+function defaultLog(request) {
+  if (request.type === 'response') {
+    log.debug({ request }, `Received response for request ${request.id}`);
+
+    return;
+  }
+
+  log.debug({ request }, `Making request ${request.id} to ${request.method} ${request.uri}`);
+}
+
+/**
+ * Exports.
+ */
+
+export default (log = defaultLog) => requestLogger(request, (request, instance) => {
+  obfuscate(request, instance);
+  log(request, instance);
```

Comment:
Yes, if the use wants to get the original request body of a response type object (example [here](https://github.com/seegno/bitcoin-core/pull/19/files/ea7b7da7e3300dde5630e8b20a0b07447ee900fe#diff-b5b0db88072fa25df71565c4843a04e5R71)).